src/config/cloudtrail.json

```diff
@@ -0,0 +1,1006 @@
+        },
+        {
+          "key": "aws.resource.id",
+          "pattern": "if( log_content_parsed.eventName == 'FailoverDBCluster', &log_content_parsed.requestParameters.dBClusterIdentifier, &EmptyValue, @ )"
+        },
+        {
+          "key": "dt.source_entity",
+          "pattern": "if( log_content_parsed.eventName == 'FailoverGlobalCluster', &dt_meid_supporting_service_v2( 'Aurora', \"aws.arn\"), &EmptyValue, @ )"
+        },
+        {
+          "key": "aws.arn",
+          "priority": 10,
+          "pattern": "if( log_content_parsed.eventName == 'FailoverGlobalCluster', &format_arn('arn:{}:rds:{}:{}:cluster:{}', [partition, region, account_id, log_content_parsed.requestParameters.globalClusterIdentifier]), &EmptyValue, @ )"
+        },
+        {
+          "key": "aws.resource.id",
+          "pattern": "if( log_content_parsed.eventName == 'FailoverGlobalCluster', &log_content_parsed.requestParameters.globalClusterIdentifier, &EmptyValue, @ )"
+        },
+        {
+          "key": "dt.source_entity",
+          "pattern": "if( log_content_parsed.eventName == 'ListTagsForResource' && contains(log_content_parsed.requestParameters.resourceName, ':cluster:'), &dt_meid_supporting_service_v2( 'Aurora',  log_content_parsed.requestParameters.resourceName), &EmptyValue, @ )"
+        },
+        {
+          "key": "dt.source_entity",
+          "pattern": "if( log_content_parsed.eventName == 'ListTagsForResource' && contains(log_content_parsed.requestParameters.resourceName, ':db:'), &dt_meid_rds_v2( log_content_parsed.requestParameters.resourceName), &EmptyValue, @ )"
+        },
+        {
+          "key": "aws.resource.id",
+          "pattern": "if( log_content_parsed.eventName == 'ListTagsForResource', &replace_regex(log_content_parsed.requestParameters.resourceName, 'arn:[^:]*:rds:[^:]*:[^:]*:[^:]*:([^:]*)','$1') &EmptyValue, @ )"
+        },
+        {
+          "key": "aws.arn",
+          "priority": 10,
+          "pattern": "if( log_content_parsed.eventName == 'ListTagsForResource', &log_content_parsed.requestParameters.resourceName, &EmptyValue, @ )"
+        },
+        {
+          "key": "dt.source_entity",
+          "pattern": "if( log_content_parsed.eventName == 'ModifyDBCluster', &dt_meid_supporting_service_v2( 'Aurora', \"aws.arn\"), &EmptyValue, @ )"
+        },
+        {
+          "key": "aws.arn",
+          "priority": 10,
+          "pattern": "if( log_content_parsed.eventName == 'ModifyDBCluster', &format_arn('arn:{}:rds:{}:{}:cluster:{}', [partition, region, account_id, log_content_parsed.requestParameters.dBClusterIdentifier]), &EmptyValue, @ )"
+        },
+        {
+          "key": "aws.resource.id",
+          "pattern": "if( log_content_parsed.eventName == 'ModifyDBCluster', &log_content_parsed.requestParameters.dBClusterIdentifier, &EmptyValue, @ )"
```

Comment:
We do need a profiling run, as simple as possible but there are few places that might be perf heavy and I think we need just basic estimation which places have strong impact and which do not

Reply:
But like really simple, 10000 x same cloudtrail log entry

Reply:
I have a setup with some thousand entries, but haven't run it yet - I plan to do it tomorrow

Reply:
First profiling result: 
I ran in first on a much smaller scale, with real data from my AWS setup 70-90 log entries that was still running 20-30 sec. I managed to get it down to 2-3s.

The story was similar to that with Grok - initialization happening too often. The default limit for number of cached/"compiled" Jmespath patterns/expressions was 128. With CloudTrail itself having 228 entries, this resulted in jmespath.parser.parse() reparsing all the patterns on for every logEntry and taking most of the execution time.

So this had nothing to do with if/switch of ==/contains ;) 

I will still do some further tests.

Reply:
It seems it is still a lot (2-3 sek for 70-90 entries)

Reply:
I've run setups with different numbers of attribute patterns, most of them resulting in execution duration exceeding Lambda timeout of 60s (both in AWS and local setup):

15k, max_size=20000 > 60s
7,5k > 55-60s
5k > 50-60s

I'm setting the cache limit deliberately high, as it seems to have no noticable memory penalty in AWS setup, whereas the penalty for not having them cached was severe. While it barely works at my test of 5k in a single config file, I still would rather leave space for any improvements elsewhere + a scenario of it being thousands of attribute patterns but in various files - rather than risk recompiling the patterns per each log entry again. 

But we can discuss how high the limit should be - I'm setting it at 20k attribute patterns for now.

Reply:
Given current number of patterns, can we ingest 10MB RDS cloudtrail logs per minute?

Reply:
No, we can't - as discussed on a meeting, we're considering it too slippery for now to go along with it. 

I'm pushing current changes to a separate branch: APM-299389-metadata-extraction-and-entity-mapping-for-cloudtrail-rds_declined
In this PR branch I'm reverting from cloudtrail config all entity mapping (leaving only the most generic CloudTrail fields that aren't event-specific) and the tests section for those. All the rest stays the same.